src/compute-types/src/plan/flat_plan.rs

```diff
@@ -761,6 +804,182 @@ impl<T> FlatPlanNode<T> {
 
         first.into_iter().chain(list).chain(last)
     }
+
+    /// Renders a single `FlatPlanNode` as a string.
+    ///
+    /// Typically of the format "{NodeName}::{Detail} {input LirID} ({options})"
+    pub fn humanize(&self, humanizer: &dyn ExprHumanizer) -> String {
```

Comment:
I think you could avoid some allocations here by declaring a mutable string and then `write!`ing at that string instead of `format!`ting.